support-frontend/assets/pages/digital-subscription-checkout/components/datePicker/datePickerHelpers.test.js

```diff
@@ -0,0 +1,62 @@
+// flow
+
+import {
+  dateIsPast,
+  dateIsOutsideRange,
+  getRange,
+  getLatestAvailableDateText,
+} from './helpers';
+import { monthText } from 'pages/paper-subscription-checkout/helpers/subsCardDays';
+
+
+describe('dateIsPast', () => {
+  it('should be able to indicate if a selected date is in the past', () => {
+    const pastDate = new Date('09/23/2020');
+    const today = Date.now();
+    const resultPast = dateIsPast(pastDate);
+    expect(resultPast).toBe(true);
+
+    const resultToday = dateIsPast(new Date(today));
+    expect(resultToday).toBe(false);
+  });
+
+});
+
+describe('dateIsOutsideRange', () => {
+  it('should be able to indicate if a selected date is too far in advance', () => {
+    const dateWithinRange = !dateIsOutsideRange(new Date(Date.now()));
+    expect(dateWithinRange).toBe(true);
+
+    const rangeDate = new Date();
+    rangeDate.setDate(rangeDate.getDate() + 93);
+    const dateOutsideRange = dateIsOutsideRange(rangeDate);
+
+    expect(dateOutsideRange).toBe(true);
+
+  });
+
+});
+
+describe('getRange', () => {
+  it('should be able to get the latest available date from now', () => {
+    const latestAvailable = getRange();
+    const rangeDate = new Date();
```

Comment:
this gets the actual date twice, and expects them to be the same.  this means the build can randomly fail e.g.
```
17:09:00
  FAIL assets/pages/digital-subscription-checkout/components/datePicker/datePickerHelpers.test.js
17:09:00
    ● getRange › should be able to get the latest available date from now
17:09:00
  17:09:00
      expect(received).toEqual(expected) // deep equality
17:09:00
  17:09:00
      Expected: 2021-01-06T17:08:59.555Z
17:09:00
      Received: 2021-01-06T17:08:59.554Z
```

I would recommend making the test deterministic by injecting the date function into getRange
Then the real code can pass in `new Date()` and the tests can pass in `new Date(2020, 10, 9)` or some other convenient hard coded date.

Reply:
I have raised a card https://trello.com/c/4l2cBx52/3393-fix-non-deterministic-test

Reply:
PR https://github.com/guardian/support-frontend/pull/2806